packages/xrpl/src/Wallet/batchSigner.ts

```diff
@@ -0,0 +1,222 @@
+}
+
+/**
+ * Takes several transactions with BatchSigners fields (in object or blob form) and creates a
+ * single transaction with all BatchSigners that then gets signed and returned.
+ *
+ * @param transactions The transactions to combine `BatchSigners` values on.
+ * @returns A single signed Transaction which has all BatchSigners from transactions within it.
+ * @throws ValidationError if:
+ * - There were no transactions given to sign
+ * @category Signing
+ */
+export function combineBatchSigners(
+  transactions: Array<Batch | string>,
+): string {
+  if (transactions.length === 0) {
+    throw new ValidationError('There are 0 transactions to combine.')
+  }
+
+  const decodedTransactions: Transaction[] = transactions.map((txOrBlob) => {
+    return getDecodedTransaction(txOrBlob)
+  })
+
+  decodedTransactions.forEach((tx) => {
+    if (tx.TransactionType !== 'Batch') {
+      throw new ValidationError('TransactionType must be `Batch`.')
+    }
+    /*
+     * This will throw a more clear error for JS users if any of the supplied transactions has incorrect formatting
+     */
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- validate does not accept Transaction type
+    validateBatch(tx as unknown as Record<string, unknown>)
+    if (tx.BatchSigners == null || tx.BatchSigners.length === 0) {
+      throw new ValidationError(
+        'For combining Batch transaction signatures, all transactions must include a BatchSigners field containing an array of signatures.',
+      )
+    }
+
+    if (tx.TxnSignature != null || tx.Signers != null) {
+      throw new ValidationError('Batch transaction must be unsigned.')
+    }
+  })
+
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- checked above
+  const batchTransactions = decodedTransactions as Batch[]
+
+  validateBatchTransactionEquivalence(batchTransactions)
+
+  return encode(getTransactionWithAllBatchSigners(batchTransactions))
+}
+
+/**
+ * The transactions should all be equal except for the 'Signers' field.
+ *
+ * @param transactions - An array of Transactions which are expected to be equal other than 'Signers'.
+ * @throws ValidationError if the transactions are not equal in any field other than 'Signers'.
+ */
+function validateBatchTransactionEquivalence(transactions: Batch[]): void {
+  const exampleTransaction = JSON.stringify({
+    flags: transactions[0].Flags,
+    transactionIDs: transactions[0].RawTransactions.map((rawTx) =>
+      hashSignedTx(rawTx.RawTransaction),
+    ),
+  })
+  if (
+    transactions.slice(1).some(
+      (tx) =>
+        JSON.stringify({
+          flags: tx.Flags,
+          transactionIDs: tx.RawTransactions.map((rawTx) =>
+            hashSignedTx(rawTx.RawTransaction),
+          ),
+        }) !== exampleTransaction,
+    )
+  ) {
+    throw new ValidationError(
+      'Flags and transaction hashes are not the same for all provided transactions.',
+    )
```

Comment:
I prefer that you compare the entire transaction payload. I understand that `flags` and `txIDs` must be identical for all equivalent Batch transactions. However, this implementation completely ignores all the other fields.

Do we take a large performance hit if we compare all the fields of the Transaction object? (except `BatchSigner`, of course)

Reply:
You're only signing the flags and transaction IDs, so that's all that matters.